test/onigumo_cli_test.exs

```diff
@@ -32,27 +33,43 @@ defmodule OnigumoCLITest do
   describe("Onigumo.CLI.main/1") do
     for argument <- @invalid_arguments do
       test("run CLI with invalid argument #{inspect(argument)}") do
-        assert usage_message_printed?(fn -> Onigumo.CLI.main([unquote(argument)]) end)
+        assert usage_message_printed?(
+                 fn -> Onigumo.CLI.main([unquote(argument)]) end,
+                 "invalid COMPONENT #{unquote(argument)}"
+               )
       end
     end
 
     test("run CLI with no arguments") do
-      assert usage_message_printed?(fn -> Onigumo.CLI.main([]) end)
+      assert usage_message_printed?(
+               fn -> Onigumo.CLI.main([]) end,
+               "exactly one COMPONENT must be provided"
+             )
     end
 
     test("run CLI with more than one argument") do
-      assert usage_message_printed?(fn -> Onigumo.CLI.main(["Downloader", "Parser"]) end)
+      assert usage_message_printed?(
+               fn -> Onigumo.CLI.main(["Downloader", "Parser"]) end,
+               "exactly one COMPONENT must be provided"
+             )
     end
 
     for switch <- @invalid_switches do
       test("run CLI with invalid switch #{inspect(switch)}") do
-        assert usage_message_printed?(fn -> Onigumo.CLI.main([unquote(switch)]) end)
+        assert usage_message_printed?(
+                 fn -> Onigumo.CLI.main(unquote(switch)) end,
+                 "invalid OPTIONS #{Enum.join(unquote(switch), ", ")}"
+               )
       end
     end
 
     for combination <- @invalid_combinations do
       test("run CLI with invalid combination #{inspect(combination)}") do
-        assert usage_message_printed?(fn -> Onigumo.CLI.main(unquote(combination)) end)
+        assert usage_message_printed?(
+                 fn -> Onigumo.CLI.main(unquote(combination)) end,
+                 # there is no easy way to do it dynamically
+                 "incompatible OPTIONS --help"
```

Comment:
```suggestion
                 "incompatible OPTIONS #{invalid}"
```

If the constant contained tuples as imagined above, the string would look like this.